src/installer.ts

```diff
@@ -338,10 +365,62 @@ async function resolveVersionFromManifest(
   }
 }
 
+function evaluateNightlyVersions(
+  versions: string[],
+  versionSpec: string
+): string {
+  let version = '';
+  let range: string | null | undefined;
+  const [raw, prerelease] = versionSpec.split('-');
+  const isValidVersion = semver.valid(raw);
+  const rawVersion = isValidVersion ? raw : semver.coerce(raw);
+  if (rawVersion) {
+    if (prerelease !== 'nightly') {
+      range = `${rawVersion}-${prerelease.replace('nightly', 'nightly.')}`;
+    } else {
+      range = `${semver.validRange(`^${rawVersion}-0`)}-0`;
+    }
+  }
+
+  if (range) {
+    versions = versions.sort((a, b) => {
+      if (semver.gt(a, b)) {
+        return 1;
+      }
+      return -1;
+    });
+    for (let i = versions.length - 1; i >= 0; i--) {
+      const potential: string = versions[i];
+      const satisfied: boolean = semver.satisfies(
+        potential.replace('-nightly', '-nightly.'),
+        range,
+        {includePrerelease: true}
+      );
+      if (satisfied) {
+        version = potential;
+        break;
+      }
+    }
```

Comment:
```suggestion
    versions.sort((a, b) => semver.lt(a, b) * 1 - 0.5);
    for (const currentVersion of versions) {
        const satisfied: boolean = semver.satisfies(
            currentVersion.replace("-nightly", "-nightly."),
            range,
            { includePrerelease: true }
          );
    if (satisfied) {
        version = currentVersion;
        break;
    }
}
```

Reply:
Uuuu @e-korolevskii  has a few aces up in his sleeve :D